libs/zglfw/src/zglfw.zig

```diff
@@ -233,6 +244,147 @@ pub const CursorMode = enum(i32) {
 };
 //--------------------------------------------------------------------------------------------------
 //
+// Joystick
+//
+//--------------------------------------------------------------------------------------------------
+
+pub const Joystick = struct {
+    pub const ButtonAction = enum(u8) {
+        release = 0,
+        press = 1,
+    };
+
+    jid: u4,
+
+    pub const maximum_supported = 16;
+
+    pub fn init(jid: u4) ?Joystick {
```

Comment:
I would prefer a global function `getJoystick(jid: u4) ?Joystick`. `init()` function suggests that `deinit()` should be also present.